fingerprint2.js

```diff
@@ -463,69 +470,25 @@
       // http://blogs.adobe.com/webplatform/2013/01/28/blending-features-in-canvas/
       // https://github.com/Modernizr/Modernizr/blob/master/feature-detects/canvas/blending.js
-      try {
-        ctx.globalCompositeOperation = "screen";
-      } catch (e) { /* squelch */ }
-      result.push("canvas blending:" + ((ctx.globalCompositeOperation === "screen") ? "yes" : "no"));
-      // detect browser support of canvas winding
-      // http://blogs.adobe.com/webplatform/2013/01/30/winding-rules-in-canvas/
-      // https://github.com/Modernizr/Modernizr/blob/master/feature-detects/canvas/winding.js
-      ctx.rect(0, 0, 10, 10);
-      ctx.rect(2, 2, 6, 6);
-      result.push("canvas winding:" + ((ctx.isPointInPath(5, 5, "evenodd") === false) ? "yes" : "no"));
-      // https://www.browserleaks.com/canvas#how-does-it-work
-      var txt = "https://github.com/valve for PEACE in Ukraine!";
-      ctx.textBaseline = "top";
-      // ios8 specific font
-      ctx.font = "72px 'DamascusLight'";
+      // https://securehomes.esat.kuleuven.be/~gacar/persistent/the_web_never_forgets.pdf
+      ctx.textBaseline = "alphabetic";
       ctx.fillStyle = "#f60";
-      ctx.fillRect(2, 0, 1000, 70);
+      ctx.fillRect(125, 1, 62, 20);
       ctx.fillStyle = "#069";
-      ctx.fillText(txt, 2, 0);
-      // android specific font
-      ctx.font = "72px 'Roboto Condensed'";
+      ctx.font = "11pt no-real-font-123";
+      ctx.fillText("Cwm fjordbank glyphs vext quiz, \ud83d\ude03", 2, 15);
       ctx.fillStyle = "rgba(102, 204, 0, 0.7)";
-      ctx.fillText(txt, 4, 2);
-      ctx.strokeStyle = "rgba(202, 104, 0, 0.9)";
-      // osx specific font
-      ctx.font = "72px 'Menlo'";
-      ctx.strokeText(txt, 8, 4);
-      // canvas blending
-      // http://blogs.adobe.com/webplatform/2013/01/28/blending-features-in-canvas/
-      // http://jsfiddle.net/NDYV8/16/
-      ctx.globalCompositeOperation = "multiply";
-      ctx.fillStyle = "rgb(255,0,255)";
-      ctx.beginPath();
-      ctx.arc(50, 50, 50, 0, Math.PI * 2, true);
-      ctx.closePath();
-      ctx.fill();
-      ctx.fillStyle = "rgb(0,255,255)";
-      ctx.beginPath();
-      ctx.arc(100, 50, 50, 0, Math.PI * 2, true);
-      ctx.closePath();
-      ctx.fill();
-      ctx.fillStyle = "rgb(255,255,0)";
-      ctx.beginPath();
-      ctx.arc(75, 100, 50, 0, Math.PI * 2, true);
-      ctx.closePath();
-      ctx.fill();
-      ctx.fillStyle = "rgb(255,0,255)";
-      // canvas winding
-      // http://blogs.adobe.com/webplatform/2013/01/30/winding-rules-in-canvas/
-      // http://jsfiddle.net/NDYV8/19/
-      ctx.arc(75, 75, 75, 0, Math.PI * 2, true);
-      ctx.arc(75, 75, 25, 0, Math.PI * 2, true);
-      ctx.fill("evenodd");
-      result.push("canvas fp:" + canvas.toDataURL());
-      return result.join("~");
+      ctx.font = "18pt Arial";
+      ctx.fillText("Cwm fjordbank glyphs vext quiz, \ud83d\ude03", 4, 45);
+      return canvas.toDataURL();
```

Comment:
1st the text fits in the rectangle. For the canvas blending and winding I removed it because they didn't talk about this in the article on canvas fingerprinting. However i can add it back.
Concerning why it should improve fingerprinting, look at the section 3.2 of this document : https://securehomes.esat.kuleuven.be/~gacar/persistent/the_web_never_forgets.pdf